### DISCUSSION TITLE
ServerError (timed out) when using NuxtJS in SSG mode with Sanity

### DISCUSSION THREAD

**Discussion by ghost** [nuxt/nuxt#10732] [Questions (Nuxt 2)] [answered]
**Describe the bug**

I have a NuxtJS (Version 2) app running in SSG mode and Sanity.io as CMS. NuxtJS generates/prerenders about 300 pages (data comes from Sanity API), which takes quite some time. The problem is that after a while I get the following error (at build stage):

```ServerError: The operation timed out```

So I guess Sanity is closing the server connection, while NuxtJS is still generating pages. I reduced the amount of pages to about 280 and then it works. So it seems like Sanity closes the connection too early.

**Possible solutions**

1. Is there a way to keep the server connection to Sanity.io longer alive, before the operation is timed out?
2. Is there a way to split up the static site generation with NuxtJS, so that after (let‘s say) 50 pages the current connection to sanity closes and a new one starts for the next 50 pages?

Is there any other solution you can think of?

**Comment by danielroe** [comment, marked as answer]
If you have a shared body of information you request on each page, consider https://nuxtjs.org/docs/configuration-glossary/configuration-generate/#speeding-up-dynamic-route-generation-with-payload.

Otherwise, you can adjust `generate.concurrency` and/or `generate.interval`.

**Comment by ghost** [comment]
Thanks! The `generate.interval` of `250` did the job in my case and did not really increase the generation time. On the other hand, adjusting `generate.concurrency` increased the generation time by a lot, because the script waits for about 30 seconds before continuing with the next group of routes.

The payload option did not change anything, because in my app I’m fetching the data in the middleware.
@danielroe So maybe fetching data in the middleware is the same as using payload? (because the middleware variant is way much easier to implement)
